src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java

```diff
@@ -370,10 +389,9 @@ private static boolean containsWhitespaceBefore(int before, int... line) {
      * @return checks if given character is valid
      */
     private static boolean isCharacterValidAfterGenericEnd(char charAfter) {
-        return charAfter == '(' || charAfter == ')'
-            || charAfter == ',' || charAfter == '['
-            || charAfter == '.' || charAfter == ':'
-            || charAfter == ';'
+        return charAfter == ')' || charAfter == ','
```

Comment:
#### Removal of `charAfter == '('`:
(As determined in https://github.com/checkstyle/checkstyle/pull/14497#discussion_r1493346388 [last paragraph])

**Purpose of removal:**
To dispose mutant (replacing `charAfter == '('` with `false`) which was able to survive due to the changes in this fix in the following line:
https://github.com/checkstyle/checkstyle/blob/84eaed72081302e94be0ef096d56bddc19bebc7f/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java#L373
<hr>

**Why the mutant survives now:**
All the valid occurrences where Generic-End(`>`) can be followed by LPAREN(`(`) are the cases where it needs to be checked that such generic is not followed by whitespace, and as of this fix we have accounted for all such occurrences, making every possible input where `>` can be followed by `(` filter through `if` block of `processSingleGeneric()`:
https://github.com/checkstyle/checkstyle/blob/e3fa404dacb70d35ff507fce573698f63e36d9f6/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java#L223-L225
Hence never reaching the `else if` block which calls `isCharacterValidAfterGenericEnd()`, which makes it possible for the mutant i.e. (replacing `charAfter == '('` with false) survive all the tests.
<hr>

**Why the mutant didn't survived before:**
Because previously Check didn't counted whitespace between generic and record header as a violation, making this the killer input where `>` could be preceded by `)` which wasn't checked for succeeding whitespace, which made it produced `illegal follow` violation, hence killing the mutant.
<hr>

**Why this mutant cannot be killed:**
To kill this mutant we would need an input where `>` is followed by `(` but isn't from ones that we have already accounted for(method, ctor, record). Regression is unable to find such input, and to the best of my knowledge, such input does not exist, but if anyone is aware of such input please do tell.
<hr>

In conclusion, it is safe to remove that operand (also provides us with nano performance gain) unless anyone is aware of the killer input or if there is any plan to use this exact method in the future where that operand would be required.
The only usage of the method with mutation is in the GenericWhitespaceCheck. (1 Usage)
GenericWhitespaceCheck does not affects any other class.